### DISCUSSION TITLE
I can't convert a variable to png format

### DISCUSSION THREAD

**Discussion by sadness-mugetsu** [SixLabors/ImageSharp#1321] [Q&A]
In General, the avatar variable takes the user's avatar in webp format, but I need it in png and for some reason I can't convert the avatar variable to png format library discord.net

```
public async Task TestCmd()
        {
            var background = SixLabors.ImageSharp.Image.Load("My way to photos");

           

            var avatar = SixLabors.ImageSharp.Image.Load(GetProfileAvatar(Context.User), out format);

            var imgpostion = new Point(x: 20, y: (background.Height / 2 - avatar.Height / 2));

            var img = background.Clone(x => 
            {
                    x.DrawImage(avatar, imgpostion, 1);
               
            });
            

            var outputFile = $"My way to photos";

            img.SaveAsPng(outputFile);

            await Context.Channel.SendFileAsync(outputFile);

        }
        public static Stream GetProfileAvatar(IUser user)
        {
            WebRequest request = WebRequest.Create(user.GetAvatarUrl());
            WebResponse response = request.GetResponse();
            Stream responseStream = response.GetResponseStream();
            return responseStream;
        }


```

**Comment by JimBobSquarePants** [comment]
>for some reason I can't convert the avatar variable to png format library discord.net

What do you mean by this? What happens?

**Reply by sadness-mugetsu** [reply]
The avatar variable takes the user's avatar but it takes the user's avatar in webp format and I need it to take png format

**Reply by JimBobSquarePants** [reply]
ImageSharp does not support webp yet.

**Reply by sadness-mugetsu** [reply]
I know that's why I want to convert the avatar variable to png format

**Reply by tocsoft** [reply]
Only way to convert a webp image to any other format would required loading it first and then re-encoding which is something ImageSharp can't currently help with I'm afraid.
